publication-rest/src/main/java/no/unit/nva/publication/update/UpdatePublicationHandler.java

```diff
@@ -167,6 +174,30 @@ protected PublicationResponse processInput(PublicationRequest input,
         return PublicationResponseFactory.create(updatedPublication, requestInfo, identityServiceClient);
     }
 
+    private Resource partialUpdate(PartialUpdatePublicationRequest request, Resource resource,
+                                   PublicationPermissions permissions, UserInstance userInstance)
+        throws ApiGatewayException {
+
+        permissions.authorize(PARTIAL_UPDATE);
+        authorizeFileEntries(resource, userInstance, getExistingFilesToUpdate(resource, request.associatedArtifacts()));
+
+        var updatedResource = request.generateUpdate(resource);
+
+        var customer = fetchCustomerOrFailWithBadGateway(customerApiClient, resource.getCustomerId());
+        var updatedFiles = updatedResource.getFiles();
+        if (!updatedFiles.stream().allMatch(file -> canUpdateFileToPendingOpenFile(file, customer, resource,
+                                                                                   updatedResource,
+                                                                                   userInstance))) {
+            throw new ForbiddenException();
+        }
+        setRrsOnFiles(updatedResource, resource, customer, userInstance.getUsername(), permissions);
+
+        new PublishingRequestResolver(resourceService, ticketService, userInstance, customer)
+            .resolve(resource, updatedResource);
+
+        return updatedResource.update(resourceService, userInstance);
+    }
```

Comment:
This method is very similar to updateMetadata() method. I think I will combine these two methods to one in next pr. Only validation differs here.